rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize.cpp

```diff
@@ -54,10 +54,114 @@ void ResizeNode::create_node()
+        adjust_out_roi_size();
+        _dst_roi_size[0] = std::min(_dst_roi_size[0], _outputs[0]->info().width());
+        _dst_roi_size[1] = std::min(_dst_roi_size[1], _outputs[0]->info().height_single());
+        _dst_roi_width_vec.push_back(_dst_roi_size[0]);
+        _dst_roi_height_vec.push_back(_dst_roi_size[1]);
+    }
+    vx_status width_status, height_status;
+    width_status = vxCopyArrayRange((vx_array)_dst_roi_width, 0, _batch_size, sizeof(vx_uint32), _dst_roi_width_vec.data(), VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    height_status = vxCopyArrayRange((vx_array)_dst_roi_height, 0, _batch_size, sizeof(vx_uint32), _dst_roi_height_vec.data(), VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    if(width_status != 0 || height_status != 0)
+        WRN("ERROR: vxCopyArrayRange _dst_roi_width or _dst_roi_height failed " + TOSTR(width_status) + "  " + TOSTR(height_status));
+    _outputs[0]->update_image_roi(_dst_roi_width_vec, _dst_roi_height_vec);
+    _dst_roi_width_vec.clear();
+    _dst_roi_height_vec.clear();
 }
 
-void ResizeNode::update_node()
-{
-  
-}
\ No newline at end of file
+void ResizeNode::init(unsigned dest_width, unsigned dest_height, RocalResizeScalingMode scaling_mode,
+                      std::vector<unsigned> max_size, RocalResizeInterpolationType interpolation_type) {
+    _scaling_mode = scaling_mode;
+    _dest_width = dest_width;
+    _dest_height = dest_height;
+    _interpolation_type = (int)interpolation_type;
+    _src_roi_size.resize(2);
+    _dst_roi_size.resize(2);
+    _max_roi_size = max_size;
+}
+
+void ResizeNode::adjust_out_roi_size() {
+    const unsigned dim = 2; // Currently supports only 2D images
+    std::vector<double> scale(dim, 1);
+    std::vector<bool> has_size(dim, false);
+    unsigned sizes_provided = 0;
+    bool has_max_size = _max_roi_size.size() > 0;
+    for (unsigned i = 0; i < dim; i++) {
+        has_size[i] = (_src_roi_size[i] != 0) && (_dst_roi_size[i] != 0);
+        sizes_provided += has_size[i];
+        scale[i] = _src_roi_size[i] ? (_dst_roi_size[i] / static_cast<double>(_src_roi_size[i])) : 1;
+    }
+    if (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_STRETCH) {
+        if (sizes_provided < dim) {
+            for (unsigned i = 0; i < dim; i++) {
+                if (!has_size[i])
+                    _dst_roi_size[i] = _src_roi_size[i];
+            }
+        }
+        if (has_max_size) {
+            for (unsigned i = 0; i < dim; i++) {
+                if ((_max_roi_size[i] > 0) && (_dst_roi_size[i] > _max_roi_size[i]))
+                    _dst_roi_size[i] = _max_roi_size[i];
+            }
+        }
+    } else if (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_DEFAULT) {
+        if (sizes_provided < dim) {
+            double average_scale = 1;
+            for (unsigned i = 0; i < dim; i++) {
+                if (has_size[i])
+                    average_scale *= scale[i];
+            }
+            if (sizes_provided > 1)
```

Comment:
There is something wrong about this logic where average_scale is computed twice

Reply:
The average scale computation comes into picture when we have more than 2 dimensions, Since we are going to restrict our code to only width and height (i.e 2-dimension) we can eliminate the average scale computation